src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,27 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch
+                {
+                    //TODO What is the correct behaviour for discarding exceptions
```

Comment:
Unless you have some logging mechanism special to serial ports, I believe we would do nothing.
If you think it should never happen, or at least not in our tests, and might indicate a bug if it did, you could add Debug.Assert.